### DISCUSSION TITLE
TF2: Better understanding the taus code using SCAM and FHIST

### DISCUSSION THREAD

**Discussion by bstephens82** [NCAR/amwg_dev#106] [TF2: Tuning]
I did some SCAM DYCOMSRF01 (stratocumulus) experiments @JulioTBacmeister was curious about, using a CAM 053 tag and the CAM 053 tag with the new CLUBB external from [@Katetc's pull request](https://github.com/ESCOMP/CAM/pull/545).  In both code bases I've turned off the l_do_expldiff_rtm_thlm flag, and in the version with the new external I've [reverted a handful of things](https://github.com/NCAR/amwg_dev/discussions/101#discussioncomment-2586729) so that the model will give essentially the same climate as the "clean" 053 code.

In these experiments I compare the DYCOMS SCAM results from the code bases described above, with l_diag_Lscale_from_tau=F, to the same experiments with the taus code turned on.  In other words there are four tests: (1) "clean" 053 without taus code, (2) "clean" 053 with taus code, (3) new external code without taus, and (4) new external code with taus.

The SCAM DYCOMS experiments only last two days, and there is no discernible difference in SWCF between the four experiments during the course of the experiments.  The cloud liquid water path variable TGCLDLWP however shows obvious differences among the four experiments:
![lwp](https://user-images.githubusercontent.com/59519056/164360282-ae3115bc-5331-4fd1-b9c1-4b1296ccb26a.png)
Clearly with the taus code turned on, but no other retuning, etc., this LWP variable takes off, which I'm guessing may be consistent with the huge increase in SWCF that we see in longer experiments with global CAM when we turn the taus code on there without retuning.

I would've guessed that the tests without the taus code would've looked more similar since the code bases can produce similar climates, but they diverge after about 3 hours, presumably due to the remaining non-reverted aspects of the new external.

**Comment by vlarson** [comment]
> The SCAM DYCOMS experiments only last two days, and there is no discernible difference in SWCF between the four experiments during the course of the experiments. The cloud liquid water path variable TGCLDLWP however shows obvious differences among the four experiments

How can the LWP change so much without the SWCF changing?

**Comment by JulioTBacmeister** [comment]
I had the same question as Vince.  We should check the DYCOMS set-up.  I
wonder what radiation is doing.

**Reply by bstephens82** [reply]
This struck me as odd as well, but I didn't know offhand if the SWCF behaves differently or is not handled the same way in SCAM. I'll try to find out what's going on this morning.

**Comment by bstephens82** [comment]
Ok, I was mistaken about the SWCF not being different---I needed to reverse the order in which the lines were plotted since the lines have a very similar shape and one was obscuring magnitude differences in the other.  Now the differences are clear:
![swcf](https://user-images.githubusercontent.com/59519056/164488323-a2d3c9ab-2058-48fe-a604-7fcd7a868e94.png)
So this looks pretty consistent with the LWP path now.  With the taus code on, they show pretty similar SWCF, similar to the very similar LWP from the above plots.  With the taus code off, the new external shows larger magnitude of SWCF, consistent with the larger LWP above.

**Comment by adamrher** [comment]
The differences in 053 clean and NEWEXT revert are weird. I'd be curious to see those runs w/ MG2 microphysics, which requires *not* setting ```-phys cam_dev```. Are these all L58?

**Reply by bstephens82** [reply]
These are actually pretty out-of-the-box SCAM runs.  I didn't know whether cam_dev or L58 would play nice with SCAM, so I just reused some earlier commands to generate these cases.  CAM_CONFIG_OPTS is "-phys cam6 -dyn eul -scam".  res=T42_T42.

**Reply by adamrher** [reply]
I see. I really don't trust SCAM, so not sure how serious to take the results. You could ask Julio if it's worth running his more trustworthy SCAM mods he developed, over the MAGIC region (MarineSc->TradeCu transition). These mods force SCAM with ERA.

**Reply by bstephens82** [reply]
Actually just as a note, I find that SCAM won't run on an L58 grid, at least not with the ncdata file I've been using (/glade/p/cesm/amwg_dev/juliob/cam_ic_files/se_dycore/FWsc_ne30pg3_58L_GRID_48_taperstart10km_lowtop_BL10_v3_beta1p75_Top_43km.nc).  It complains about not finding a longitude coordinate I believe.

It will run with CAM_CONFIG_OPTS="-phys cam_dev" though.

**Reply by adamrher** [reply]
if you want to run L58 w/ SCAM use this ncdata file:
```
/glade/work/aherring/grids/vertical-res/L58/SCAM_IC_288x192_L58_48_BL10.nc
```

But you'll have to comment out around line 845 of src/dynamics/eul/dyn_comp.F90:
```
!+++ARH
!            call endrun(sub//': ERROR:  Units for tracer ' &
!               //trim(cnst_name(m_cnst))//' must be in KG/KG')
!---ARH
```
Since the ncdata file is missing units for some of the variables.

**Reply by bstephens82** [reply]
Ah ok, thanks.  I'll see if Julio would like to see those experiments.  I'm about the post the plots for the cam_dev experiments here in a sec.

**Comment by bstephens82** [comment]
Just for reference, here is what I get for LWP and SWCF for the same set of experiments when I run the same tests with -phys cam_dev (still on a 32-level grid).
![lwp_camdev](https://user-images.githubusercontent.com/59519056/164512575-0cf1d0d1-b770-42de-a99c-8b81621ac52e.png)
![swcf_camdev](https://user-images.githubusercontent.com/59519056/164512590-d281a167-b162-4433-bbc5-78af96bb2644.png)

**Comment by bstephens82** [comment]
I thought I'd just quickly run the same SCAM experiments with cam_dev and the L58 ncdata file Adam suggested above.  Interestingly the LWP climbs now for all cases:
![lwp_camdev_l58adam](https://user-images.githubusercontent.com/59519056/164532278-cb98ec5c-815d-45d4-a41f-9e1889460a7e.png)
![swcf_camdev_l58adam](https://user-images.githubusercontent.com/59519056/164532296-c541e3da-6851-433e-a477-cfbab550d1c0.png)

**Comment by bstephens82** [comment]
@JulioTBacmeister and @adamrher Here are a few sets of diagnostics that show the effects of using Colin's flags and parameters with some attention to the l_e3sm_config flag.  These are all 14-month, FHIST, SE core, 58-level, cam_dev runs with Kate's PR code, using Colin's parameters, although the flag settings are slightly different.

Colin's flag settings are
l_diag_Lscale_from_tau = T
l_e3sm_config = T
l_min_xp2_from_corr_wx = T
l_min_wp2_from_corr_wx = T
l_predict_upwp_vpwp = T

Using all of Colin's parameters *and* flags:
https://webext.cgd.ucar.edu/FHIST/fhist_colin_params_and_flags/atm/fhist_colin_params_and_flags-obs.1979_1979/

As you can see using all of Colin's parameters and flags gives a halfway decent looking climate with the taus code, at least SWCF is not massively over- or under-predicted.

Using all of Colin's parameters and flag *except* for the prognostic momentum or the e3sm_config flag:
https://webext.cgd.ucar.edu/FHIST/fhist_ldiagT_lminT_and_colinparams/atm/fhist_ldiagT_lminT_and_colinparams-obs.1979_1979/

This actually gives an over-predicted (ie too positive) SWCF, as compared to simply turning on the taus code with no further parameter or flag changes, which hugely under-predicts SWCF.

Using all of Colin's parameters and flags *except* for the prognostic momentum (ie this one is identical to the previous except it has the e3sm_config flag turned on):
https://webext.cgd.ucar.edu/FHIST/fhist_ldiagTe3sm_lminT_and_colinparams/atm/fhist_ldiagTe3sm_lminT_and_colinparams-obs.1979_1979/

As you can see the l_e3sm_config flag makes a big difference and gets us most of the way back to the result using all of Colin's parameters and flags (the remaining difference must be in using prognostic momentum fluxes).

**Reply by vlarson** [reply]
> @JulioTBacmeister and @adamrher Here are a few sets of diagnostics that show the effects of using Colin's flags and parameters with some attention to the l_e3sm_config flag.  These are all 14-month, FHIST, SE core, 58-level, cam_dev runs with Kate's PR code, 
> Using all of Colin's parameters and flag _except_ for the prognostic momentum or the e3sm_config flag: [webext.cgd.ucar.edu/FHIST/fhist_ldiagT_lminT_and_colinparams/atm/fhist_ldiagT_lminT_and_colinparams-obs.1979_1979](https://webext.cgd.ucar.edu/FHIST/fhist_ldiagT_lminT_and_colinparams/atm/fhist_ldiagT_lminT_and_colinparams-obs.1979_1979/)
> 
> This actually gives an over-predicted (ie too positive) SWCF, as compared to simply turning on the taus code with no further parameter or flag changes, which hugely under-predicts SWCF.

If you want to use Colin's parameter and flag values, and use `l_e3sm_config=F`, then it might help to remove the 0.1 in
[this line](https://github.com/larson-group/clubb_release/blob/a2bd9f0b8c90498e2e7dfc9c514934e5a7696f99/src/CLUBB_core/mixing_length.F90#L1404).  I'm guessing here, but to me that looks like the biggest difference between `l_e3sm_config=T` and `l_e3sm_config=F`.

**Reply by bstephens82** [reply]
Good guess!  I actually verified this experimentally yesterday.  Surprisingly that one little factor of 0.1 in the definition of invrs_tau_xp2_zm makes a huge difference to climate---according to my tests it changes the annual mean SWCF by 10 W/m2.  I'm continuing to explore it and will share more on Monday.

**Reply by bstephens82** [reply]
I find that if I run 14-month CAM tests with an incrementally increased "weight" on the background ("bkgnd") term of invrs_tau_xp2_zm, as you pointed to with your link, then I get a more or less linearly increasing SWCF effect, which I'm trying to understand.  Here is a table of my results:
| Weight on invrs_tau_xp2_zm "bkgnd" term | Annual mean SWCF (W/m2) |
| --- | --- |
| 0.1 | -42.50 | 
| 0.25 | -44.56 |
| 0.5 | -47.44 |
| 0.75 | -49.89 |
| 1.0 | -52.45 |

In [Zhun's paper](https://agupubs.onlinelibrary.wiley.com/doi/full/10.1029/2020MS002278), the idea is that increasing the damping on the scalar fluxes relative to the scalar variances, particularly in the cloud-top region, can lead to bright Sc but also sufficiently dim shallow Cu, hence the idea of separating out the time scales for different types of damping effects (eg damping due to pressure fluctuations vs damping due to turbulent cascade).  In this case however I'm just uniformly increasing the damping on the scalar variances over the whole column, ie just shifting the invrs_tau_xp2_zm plot to the right, and it's not obvious to me why that would lead to this increased SWCF, since based on Zhun's paper I would've associated greater damping on the scalar variances with possibly less chance of cloudiness.  @vlarson Do you have an idea why this might be so?

**Reply by vlarson** [reply]
> I find that if I run 14-month CAM tests with an incrementally increased "weight" on the background ("bkgnd") term of invrs_tau_xp2_zm, as you pointed to with your link, then I get a more or less linearly increasing SWCF effect, which I'm trying to understand.
> I'm just uniformly increasing the damping on the scalar variances over the whole column, ie just shifting the invrs_tau_xp2_zm plot to the right, and it's not obvious to me why that would lead to this increased SWCF, since based on Zhun's paper I would've associated greater damping on the scalar variances with possibly less chance of cloudiness. 

I'm going to guess that if xp2 (i.e., rtp2 and thlp2) is damped more, then the layer will become more well mixed, and well-mixed layers are more likely to be overcast or clear, instead of partly cloudy.  In this case, perhaps the stratocumuli are being enhanced.  In contrast, cumulus clouds require lots of variability.

**Comment by vlarson** [comment]
Our goal is to turn on these two flags: 
```
l_predict_upwp_vpwp          = .true.,
l_diag_Lscale_from_tau       = .true.,
```

However, there are some other flags that hopefully won't affect the climate too much but for various reasons ought to be set:
```
l_vert_avg_closure           = .false.
l_min_wp2_from_corr_wx       = .false.
l_min_xp2_from_corr_wx       = .false.
l_enable_relaxed_clipping    = .true.
l_smooth_Heaviside_tau_wpxp  = .true.
ipdf_call_placement          = 2
l_godunov_upwind_wpxp_ta     = .true.
l_godunov_upwind_xpyp_ta     = .true.
l_partial_upwind_wp3         = .true.
```

These flag settings might make a bigger difference to the answers:
```
l_vary_convect_depth         = .true.
l_e3sm_config                = .false.
l_standard_term_ta           = .true.
l_do_expldiff_rtm_thlm       = .false.
l_diffuse_rtm_and_thlm       = .false.
```

Experiments:
```
l_lmm_stepping               = .false.
```

**Reply by zarzycki** [reply]
Putting this here for now, although we may want to open a new ticket if it involves more than a "yes." Based on the above comment, for my current "tuning" runs, I am running...

```
clubb_l_diag_lscale_from_tau=.true.
clubb_l_predict_upwp_vpwp=.true.

clubb_l_e3sm_config=.false.
clubb_l_vert_avg_closure           = .false.
clubb_l_min_wp2_from_corr_wx       = .false.
clubb_l_min_xp2_from_corr_wx       = .false.
clubb_l_smooth_Heaviside_tau_wpxp  = .true.
clubb_ipdf_call_placement          = 2
clubb_l_godunov_upwind_wpxp_ta     = .true.
clubb_l_godunov_upwind_xpyp_ta     = .true.
clubb_l_do_expldiff_rtm_thlm       = .false.
clubb_l_stability_correct_tau_zm   = .true.
clubb_l_trapezoidal_rule_zm             =  .true.
clubb_l_trapezoidal_rule_zt             =  .true.
```

Then I have SourceMod'ed two CLUBB files (model_flags and advance_xm_wpxp_module) such that:

```
l_standard_term_ta = .true.        !model_flags
l_partial_upwind_wp3 = .true.     !model_flags
l_vary_convect_depth = .true.     !model_flags
l_enable_relaxed_clipping = .true.      !advance_xm_wpxp_module
```

So...

A) @vlarson and @bstephens82 would you recommend this deck for developing a functional CLUBBx-taus config?

If yes...

B) We should consider making those other terms either nl parameters (probably preferable) or flipping them in CLUBB source.

**Reply by bstephens82** [reply]
I believe we want clubb_l_stability_correct_tau_zm = .false. if we are using l_diag_Lscale_from_tau=.true.  You can see in advance_clubb_core_module that this affects what taus are used for some terms.  In particular, I think we want to use invrs_tau_wpxp_zm for the C6 term, and this will only happen if l_stability_correct_tau_zm=F.

Also I believe all four of the flags you mentioned at the end of your post are now namelist parameters in the Larson group CLUBB code, so the next time we update the external those could all be made into namelist parameters in CAM.

**Reply by adamrher** [reply]
I defer to Ben ;) Right now we are keeping the trap rules on so that's good you've kept them on. ```clubb_l_vert_avg_closure``` is broken and doesn't do anything (when T, it's supposed to turn both trap rules on and the ```clubb_l_call_pdf_closure_twice``` ... so just make sure this last one is true ... it defaults to true so should be).

In general I'm in favor of ```l_vary_convect_depth``` = T. We will have to remember to SourceMod these logicals when using the tunings Colin is preparing. @megandevlan and I have done some analysis of this new functionality (which primarily diagnoses the surface fluxes of the quadratic terms from the CAM surface fluxes, using a bit more sophisticated method for calculating the depth of the surface layer ... currently it's a constant like 1 meter or something), and it does make a pretty noticeable difference to the global surface fluxes. This whole bit will be moot if/when we bring in CLASP, which puts this calculation of the fluxes of the quadratic terms  into the land model where it belongs.

**Reply by megandevlan** [reply]
@adamrher your memory's better than mine! I had nearly forgotten about this, but yes `l_vary_convect_depth` had a noticeable impact on surface fluxes (and surface temperature, I believe), and I would agree that it's useful to turn on. 

CLASP can handle moving these calculations into the land model, though we'll need to consider how to integrate this with the other surface models that don't have any kind of CLASP-like calculations at the moment; I wonder if there would be some weird discontinuities at coastal boundaries without it. My current global tests just rely on the land fraction being nearly 1 in order to determine if the CLM-based boundary conditions should be used, so it needs some work to refine that further.

**Reply by zarzycki** [reply]
Alright sounds good, thanks all for the input! For potential future namelist_defaults.xml reference, the current deck I am working with is therefore:

```
clubb_l_diag_lscale_from_tau=.true.
clubb_l_predict_upwp_vpwp=.true.
clubb_l_e3sm_config=.false.
clubb_l_vert_avg_closure           = .false.
clubb_l_min_wp2_from_corr_wx       = .false.
clubb_l_min_xp2_from_corr_wx       = .false.
clubb_l_smooth_Heaviside_tau_wpxp  = .true.
clubb_ipdf_call_placement          = 2
clubb_l_godunov_upwind_wpxp_ta     = .true.
clubb_l_godunov_upwind_xpyp_ta     = .true.
clubb_l_do_expldiff_rtm_thlm       = .false.
clubb_l_stability_correct_tau_zm   = .false.
clubb_l_trapezoidal_rule_zm             =  .true.
clubb_l_trapezoidal_rule_zt             =  .true.

l_standard_term_ta = .true.        !model_flags
l_partial_upwind_wp3 = .true.     !model_flags
l_vary_convect_depth = .true.     !model_flags
l_enable_relaxed_clipping = .true.      !advance_xm_wpxp_module
```

**Comment by bstephens82** [comment]
To visualize what happens to the damping coefficients on the pr1 and dp1-type CLUBB terms with the taus code on or off, I output the values of these coefficients to a file at each timestep and created some time-height plots.  The plotted damping coefficients are `C1*invrs_tau_C1_zm, C2rt*invrs_tau_xp2_zm, C4*invrs_tau_wp2_zm, C6rt_Skw_fnc*invrs_tau_C1_zm, C8*invrs_tau_wp3_zt, C14*invrs_tau_wp2_zm`.  So if the taus code is turned off, all of the invrs_tau_* factors are the same (up to zm or zt levels) but the "C" coefficients may differ.  If the taus code is turned on, the invrs_tau_* terms differ in how they are defined.  

Anyway these plots may be useful and here is an example of one. Note that the surface values tend to throw off the color scales if they are included as they are here.

taus on (with Colin's parameters) vs taus off (default parameters):
![damping_coefs](https://user-images.githubusercontent.com/59519056/170118099-89294159-a986-4db5-89a8-c232b9e37c5d.png)

Some takeaways include the fact that the damping coefficients are typically smaller when using the taus code with Colin's parameters, and the fact that the damping coefficient with invrs_tau_wp3_zt is larger due to the presence of the additional term [`C_invrs_tau_N2_clear_wp3 * brunt_freq_out_cloud`](https://github.com/larson-group/clubb_release/blob/9b6064d7632c7d094d0afd5b7e088b8da869601e/src/CLUBB_core/mixing_length.F90#L1526) (this is something I tested separately by setting C_invrs_tau_N2_clear_wp3=0).  This is the only term that currently uses `brunt_freq_out_cloud` (invrs_tau_wpxp_zm is defined with `brunt_freq_out_cloud` but is not currently used anywhere).

**Reply by adamrher** [reply]
Interesting. Also when I look closely at the taus code off plots, there are some high-frequency oscillations that may be numerical in nature. In contrast, the c8*invrs_tau_wp3_zt "taus on" plot has these same diurnal pulses as taus off, but w/o high frequency oscillations. Staying on this c8 plot, which controls dissipation on skewness (and which Jiang has found to be important for reigning in the runaway greenhouse at 6xCO2), it looks like "taus on" has smaller magnitudes at the surface, but larger magnitudes aloft, compared to taus off. 

What is the height scale on the y-axis?

**Reply by bstephens82** [reply]
Yes sorry about the missing axis labels.  The vertical axis is just the vertical index (from 0 to 58), and the horizontal axis is time.  There must be a way to convert the vertical axis to height or pressure units, although I've struggled to do that.  Interesting catch on the high-frequency oscillations, I will look more closely at this.

**Reply by bstephens82** [reply]
I should've also pointed out that, as a first effort at looking at this output, I'm using FSCAM runs here with cam6 physics.  @adamrher do you think the oscillations could be caused by the previously-discovered process-ordering issue?

**Reply by adamrher** [reply]
yeah they may look better in cam_dev. The oscillations must be in the tau term which is L/sqrt(TKE)?

**Reply by bstephens82** [reply]
> The oscillations must be in the tau term which is L/sqrt(TKE)?

I think this is probably right that the oscillations are coming from the tau factors, which if the taus code is turned off is equal to tau=L/sqrt(TKE).  Although, it may be worth noting that sometimes the "C" parameters are used to calculate a skewness-dependent function which then acts as the numerator of the damping coefficient, as in this [piece of code](https://github.com/larson-group/clubb_release/blob/9b6064d7632c7d094d0afd5b7e088b8da869601e/src/CLUBB_core/advance_xm_wpxp_module.F90#L538-L539).  But these oscillations are general enough that I think they must be coming from the tau factors.

But also, I find there are oscillations present in both the taus-on and taus-off tests.  This plot is from the same set of experiments but is only showing the damping coefficients near the surface (k=2 out of 59 levels---these are printed directly from CLUBB core so height increases with the index).  Note that the strange behavior for the "tau_xp2" plot is coming from [this merge statement in mixing_length.F90](https://github.com/larson-group/clubb_release/blob/9b6064d7632c7d094d0afd5b7e088b8da869601e/src/CLUBB_core/mixing_length.F90#L1476-L1478).

Sorry for the cut off titles...left column is taus code ON and right column is taus code OFF.   
![damping_coefs](https://user-images.githubusercontent.com/59519056/170142099-703907de-ffb9-4025-819e-7bc8dfdc689b.png)

**Reply by adamrher** [reply]
In the "taus on" code I suspect it's the 1/tau calculation which has Ri number terms which may have the oscillations. 

This isn't the first time you've shown these oscillations in SCAM, and I'm beginning to wonder whether they are an artifact of SCAM. In SCAM, the dycore does a hard reset of the winds to the value on the iop file, every time the dycore is called, and then clubb alters these winds.

**Reply by bstephens82** [reply]
> In the "taus on" code I suspect it's the 1/tau calculation which has Ri number terms which may have the oscillations.

Maybe I'm missing something, but looking through the code it appears that invrs_tau_wpxp_zm is the only inverse tau term that depends on Ri, and I don't think that invrs_tau is currently used in the code.

**Reply by adamrher** [reply]
ok. I'll amend my speculation, that the 1/tau oscillations would arise if it has any dependency on the mean wind. For example, du/dz. Which I suspect it does.

Backing up a bit, I'm a little weary of spending time trying to chase down these oscillations in SCAM. SCAM treats momentum in weird ways, by which I mean the dycore just resets the winds to that on the iop file every time-step. So the winds in the history fields are some constant wind in the iop file, but modified by CLUBB. This may itself lead to spurious oscillations, and so I'd want to verify these occur in the 3D sims before taking them too seriously.

**Reply by bstephens82** [reply]
Yeah, there is dependence on the mean wind in most if not all the invrs_taus.  

Yeah I think it's reasonable to see if there are oscillations in global tests before trying to chase that down.  That's what I want to do next, so I'll update here when I have something to show.

**Comment by kylemnardi** [comment]
@vlarson @zarzycki Here are the tau settings I used for an experimental CAM6-CLUBB configuration (x1008) with increased influence of the buoyancy and shear terms and decreased influence of the background term (this was run for a 3-day hindcast initialized using ERA5):

``` 
C_invrs_tau_bkgnd            =   0.499999999613144
C_invrs_tau_sfc              =   0.100000000000000
C_invrs_tau_shear            =   0.200000000154742
C_invrs_tau_N2               =   0.319999999764285
C_invrs_tau_N2_wp2           =   0.100000000000000
C_invrs_tau_N2_xp2           =   5.000000000000000E-002
C_invrs_tau_N2_wpxp          =   0.000000000000000E+000
C_invrs_tau_N2_clear_wp3     =    1.00000000000000
C_invrs_tau_wpxp_Ri          =   0.350000000000000
C_invrs_tau_wpxp_N2_thresh   =   3.300000000000000E-004
```

For reference, these are the settings we're currently using as our "benchmark", based on Colin's recommendations from a few months ago:

```
C_invrs_tau_bkgnd            =    3.72712375569842
C_invrs_tau_sfc              =   0.127430725682505
C_invrs_tau_shear            =   0.125027263047880
C_invrs_tau_N2               =   8.122667220610320E-002
C_invrs_tau_N2_wp2           =   0.100000000000000
C_invrs_tau_N2_xp2           =   5.000000000000000E-002
C_invrs_tau_N2_wpxp          =   0.000000000000000E+000
C_invrs_tau_N2_clear_wp3     =    1.00000000000000
C_invrs_tau_wpxp_Ri          =   0.350000000000000
C_invrs_tau_wpxp_N2_thresh   =   3.300000000000000E-004
```

This plot shows the inverse tau profiles of the "benchmark" (solid lines) and the experimental config x1008 (dashed lines) off South America's western coast:

![maritime_stratocu_avg_2020 x1008_vert_prof_invrs_tau](https://user-images.githubusercontent.com/47332275/228038458-f5049f51-0929-46cf-af96-0c7fcb65bdb4.png)

For reference, here's the profile of the same config but with the taus code turned off (focus on the gold dashed line, which is the inverse of `tau_zm` with the taus code off...the solid lines are still from the "benchmark" with the taus code on):

![maritime_stratocu_avg_2020 x1001_vert_prof_invrs_tau](https://user-images.githubusercontent.com/47332275/228038728-0b52dc41-62ac-4e9e-9f5e-7a995718a19f.png)

**Reply by vlarson** [reply]
Thanks, Kyle.  Reducing C_invrs_tau_bkgnd sounds like a good idea.  We haven't explored values that deviate much from 1, but lower values sound more plausible than higher values.  The values in Zhun Guo's paper are similar to your revised values, except `c_invrs_tau_N2_Ri=3`:

```
   0:  C1                           =    1.00000000000000
   0:  C1b                          =    1.00000000000000
   0:  C1c                          =   0.750000000000000
   0:  C2                           =    1.30000000000000
   0:  C2b                          =    1.30000000000000
   0:  C2c                          =    5.00000000000000
   0:  C2rt                         =    1.00000000000000
   0:  C2thl                        =    1.00000000000000
   0:  C2rtthl                      =    1.00000000000000
   0:  C4                           =    1.00000000000000
   0:  C5                           =   0.000000000000000E+000
   0:  C6rt                         =    1.00000000000000
   0:  C6rtb                        =    1.00000000000000
   0:  C6rtc                        =   0.500000000000000
   0:  C6thl                        =    1.00000000000000
   0:  C6thlb                       =    1.00000000000000
   0:  C6thlc                       =   0.500000000000000
   0:  C7                           =   0.700000000000000
   0:  C7b                          =   0.700000000000000
   0:  C7c                          =   0.500000000000000
   0:  C8                           =   0.500000000000000
   0:  C8b                          =   0.000000000000000E+000
   0:  C10                          =    3.30000000000000
   0:  C11                          =   0.500000000000000
   0:  C11b                         =   0.500000000000000
   0:  C11c                         =   0.850000000000000
   0:  C12                          =    1.00000000000000
   0:  C13                          =   0.100000000000000
   0:  C14                          =    1.00000000000000
   0:  C15                          =   0.500000000000000
   0:  C_wp2_splat                  =   0.000000000000000E+000
   0:  C6rt_Lscale0                 =    14.0000000000000
   0:  C6thl_Lscale0                =    14.0000000000000
   0:  C7_Lscale0                   =   0.850000000000000
   0:  wpxp_L_thresh                =    100.000000000000
   0:  c_K                          =   0.200000000000000
   0:  c_K1                         =    1.00000000000000
   0:  nu1                          =    1.00000000000000
   0:  c_K2                         =   0.100000000000000
   0:  nu2                          =    1.00000000000000
   0:  c_K6                         =   0.375000000000000
   0:  nu6                          =    5.00000000000000
   0:  c_K8                         =    10.0000000000000
   0:  nu8                          =    60.0000000000000
   0:  c_K9                         =   0.250000000000000
   0:  nu9                          =    20.0000000000000
   0:  nu10                         =   0.000000000000000E+000
   0:  c_K_hm                       =   0.750000000000000
   0:  c_K_hmb                      =   0.100000000000000
   0:  K_hm_min_coef                =   0.100000000000000
   0:  nu_hm                        =    1.50000000000000
   0:  slope_coef_spread_DG_means_w =    21.0000000000000
   0:  pdf_component_stdev_factor_w =    6.50000000000000
   0:  coef_spread_DG_means_rt      =   0.800000000000000
   0:  coef_spread_DG_means_thl     =   0.800000000000000
   0:  gamma_coef                   =   0.30000000000000
   0:  gamma_coefb                  =   0.30000000000000
   0:  gamma_coefc                  =    1.20000000000000
   0:  mu                           =   5.000000000000000E-004
   0:  beta                         =    2.00000000000000
   0:  lmin_coef                    =   0.500000000000000
   0:  omicron                      =   0.800000000000000
   0:  zeta_vrnce_rat               =   0.000000000000000E+000
   0:  upsilon_precip_frac_rat      =   0.900000000000000
   0:  lambda0_stability_coef       =   3.000000000000000E-002
   0:  mult_coef                    =   0.500000000000000
 clubb_c_invrs_tau_bkgnd                = 1.5
 clubb_c_invrs_tau_n2                   = 0.65
 clubb_c_invrs_tau_n2_clear_wp3         = 2.
 clubb_c_invrs_tau_n2_wp2               = 0.2
 clubb_c_invrs_tau_n2_wpxp              = 2.
 clubb_c_invrs_tau_n2_xp2               = 0.0
 clubb_c_invrs_tau_sfc                  = .3
 clubb_c_invrs_tau_shear                = 0.15
 c_invrs tau_N2_thres                   = 0.00033   (not tunable in Zhun's paper)
 c_invrs tau_N2_Ri                        =  3          (not tunable in Zhun's paper)
 alt_threshold=300.
```
